### PR TITLE
SQL: Fix BasicFormatter NPE

### PR DESCRIPTION
**Pull request by astefan**
Lacking tests for this made the bug go through and discovered by manual testing. This PR should fix the bug and, also, adds tests. Bug introduced with https://github.com/elastic/elasticsearch/pull/37602

### CODE REVIEW

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/action/BasicFormatterTests.java [hunk 1]
```diff
@@ -40,14 +42,18 @@
...
         assertThat(result, arrayWithSize(4));
-        assertEquals("      foo      |         bar          |15charwidename!|superduperwidename!!!|      baz      |"
-                + "          date          ", result[0]);
-        assertEquals("---------------+----------------------+---------------+---------------------+---------------+"
-                + "------------------------", result[1]);
-        assertEquals("15charwidedata!|1                     |6.888          |12                   |rabbit         |"
-                + "1953-09-02T00:00:00.000Z", result[2]);
-        assertEquals("dog            |1.7976931348623157E308|123124.888     |9912                 |goat           |"
-                + "2000-03-15T21:34:37.443Z", result[3]);
+        logger.info(result[0]);
```
Comment (matriv) [inline x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/action/BasicFormatterTests.java]: Do we need those?

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/action/BasicFormatterTests.java [hunk 2]
```diff
@@ -57,13 +63,15 @@ public void testFormatWithHeader() {
...
-                        Arrays.asList("ohnotruncateddata", 4, 1, 77, "wombat", "1955-01-21T01:02:03.342Z"),
-                        Arrays.asList("dog", 2, 123124.888, 9912, "goat", "2231-12-31T23:59:59.999Z"))).split("\n");
+                        Arrays.asList("ohnotruncateddata", 4, 1, null, 77, "wombat", "1955-01-21T01:02:03.342Z", null),
+                        Arrays.asList("dog", 2, 123124.888, null, 9912, "goat", "2231-12-31T23:59:59.999Z", null))).split("\n");
         assertThat(result, arrayWithSize(2));
-        assertEquals("ohnotruncatedd~|4                     |1              |77                   |wombat         |"
-                + "1955-01-21T01:02:03.342Z", result[0]);
-        assertEquals("dog            |2                     |123124.888     |9912                 |goat           |"
-                + "2231-12-31T23:59:59.999Z", result[1]);
+        logger.info(result[0]);
```
Comment (matriv) [inline x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/action/BasicFormatterTests.java]: same here

**Review by matriv — APPROVED** [review]
LGTM. Left 2 comments about logging in tests.

**Review by costin — APPROVED** [review]

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/action/BasicFormatterTests.java (hunk 1 above)
Comment (astefan) [reply x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/action/BasicFormatterTests.java]: We don't. Not paying attention. Thanks for catching.

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/1

**Pull request merged**